### ISSUE TITLE
xcaddy not picking up v2.0.0

### ISSUE THREAD

**Issue by samcro1967** [kirsch33/realip#12]
v2.0.0 is not being picked up by xcaddy when a new image is created.  Build logs and Dockerfile are below.

xcaddy logs
```
Step 3/11 : RUN xcaddy build     --with github.com/lucaslorentz/caddy-docker-proxy/plugin/v2     --with github.com/greenpau/caddy-security     --with github.com/greenpau/caddy-trace     --with github.com/porech/caddy-maxmind-geolocation     --with github.com/caddy-dns/cloudflare     --with github.com/kirsch33/realip
 ---> Running in 846689b45504
[91m2022/05/18 06:30:31 [INFO] Temporary folder: /tmp/buildenv_2022-05-18-0630.3388528306
2022/05/18 06:30:31 [INFO] Writing main module: /tmp/buildenv_2022-05-18-0630.3388528306/main.go
package main

import (
	caddycmd "github.com/caddyserver/caddy/v2/cmd"

	// plug in Caddy modules here
	_ "github.com/caddyserver/caddy/v2/modules/standard"
	_ "github.com/lucaslorentz/caddy-docker-proxy/plugin/v2"
	_ "github.com/greenpau/caddy-security"
	_ "github.com/greenpau/caddy-trace"
	_ "github.com/porech/caddy-maxmind-geolocation"
	_ "github.com/caddy-dns/cloudflare"
	_ "github.com/kirsch33/realip"
)
...
[0m[91m2022/05/18 06:32:41 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/kirsch33/realip github.com/caddyserver/caddy/v2@v2.5.1 
[0m[91mgo: downloading github.com/kirsch33/realip v1.5.4
[0m[91mgo: added github.com/kirsch33/realip v1.5.4
```

**DockerfIle**
```
<html>
<body>
<!--StartFragment-->

ARG CADDY_VERSION=2.5.1
--
  |  
  | FROM caddy:${CADDY_VERSION}-builder AS builder
  | #FROM caddy:builder AS builder
  |  
  | RUN xcaddy build \
  | --with github.com/lucaslorentz/caddy-docker-proxy/plugin/v2 \
  | --with github.com/greenpau/caddy-security \
  | --with github.com/greenpau/caddy-trace \
  | --with github.com/porech/caddy-maxmind-geolocation \
  | --with github.com/caddy-dns/cloudflare \
  | --with github.com/kirsch33/realip
  |  
  | FROM caddy:${CADDY_VERSION}-alpine
  | #FROM caddy:builder-alpine
  |  
  | RUN apk add --no-cache tzdata
  |  
  | COPY --from=builder /usr/bin/caddy /usr/bin/caddy
  |  
  | CMD ["caddy", "docker-proxy"]

<!--EndFragment-->
</body>
</html>
```

**Comment by kirsch33** [comment]
@samcro1967 
not sure why its not picking up the newest since I released and tagged the changes to 2.0.0, however a workaround might be to try changing this Dockerfile line:
`--with github.com/kirsch33/realip`

to
`--with github.com/kirsch33/realip@latest`

alternatively, either of these should also work:
`--with github.com/kirsch33/realip@v2.0.0`
`--with github.com/kirsch33/realip@3eaa4a63330b8d23d153e555da8797865ae1b22e`

**Comment by francislavoie** [comment]
When you tag a v2 of a repo, you need to change the module name to add `/v2` at the end, I think. https://donatstudios.com/Go-v2-Modules

**Comment by samcro1967** [comment]
The build failed adding /v2.

Seeing similiar the same issue with '--with github.com/lucaslorentz/caddy-docker-proxy/plugin/v2/ in Dockerfile also.  It is not picking up the latest version 2.6.0.  Caddy ends up with v2.0.0 when rebuilt.  It is not picking up the latest version.

```
go: downloading github.com/kirsch33/realip v1.5.4
go: module github.com/kirsch33/realip@upgrade found (v1.5.4), but does not contain package github.com/kirsch33/realip/v2
```

**Comment by francislavoie** [comment]
To be clear I mean the actual repo needs an update to add `/v2`, _and then_ `/v2` will work from then-on.

https://github.com/kirsch33/realip/blob/3eaa4a63330b8d23d153e555da8797865ae1b22e/go.mod#L1

**Comment by kirsch33** [comment]
@francislavoie 
thanks i'll look into that

@samcro1967 
use the syntax i posted above and report back

**Comment by samcro1967** [comment]
@kirsch33 Neither @v2.0.0 or @latest worked for me.

**Comment by francislavoie** [comment]
The other option is to remove the v2.0.0 tag and re-tag it as a v1.x.x version instead, as long as you don't want to change the module name.

**Comment by samcro1967** [comment]
@kirsch33 Just checking to see if there is any update or anything else you wanted me to try.

**Comment by kirsch33** [comment]
@samcro1967 just implemented the versioning fix suggested by @francislavoie

tagged and released as v2.0.1

using either `--with github.com/kirsch33/realip@latest` or `--with github.com/kirsch33/realip@v2.0.1` should work now, please let me know

**Comment by samcro1967** [comment]
@latest did not work, but @v.2.0.1 did.  The build logs are below for @latest.  It is a bit weird as it seems to recognize @latest is 2.5.1, but then downloads and adds v1.5.4.

```
2022/05/25 01:59:56 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/kirsch33/realip@latest github.com/caddyserver/caddy/v2@v2.5.1 
go: downloading github.com/kirsch33/realip v1.5.4
go: added github.com/kirsch33/realip v1.5.4
```

**Comment by francislavoie** [comment]
You must add `/v2` to the module ID in your `--with`

With v2.0.1 being released and the module ID being changed, it's as if it's a totally separate package. You can't continue to use the old module ID.

Like I said earlier the other option was to re-tag as v1.x.x instead of bumping to v2.x.x and the old module ID could continue to be used.

That's just how go modules work. It's super annoying, I know. But it is what it is.

**Comment by samcro1967** [comment]
So like this `--with github.com/kirsch33/realip/v2@latest`?

**Comment by francislavoie** [comment]
Yes but you can remove `@latest`, it's redundant.

**Comment by samcro1967** [comment]
`--with github.com/kirsch33/realip/v2` worked.  It looks like it downloaded both versions for some reason, but only added 2.0.1.

```
2022/05/25 02:22:03 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/kirsch33/realip/v2 github.com/caddyserver/caddy/v2@v2.5.1 
go: downloading github.com/kirsch33/realip/v2 v2.0.1
go: downloading github.com/kirsch33/realip v1.5.4
go: added github.com/kirsch33/realip/v2 v2.0.1
```

**Comment by kirsch33** [comment]
honestly, i had no idea going to a version 2.x would cause this much headache for people. im considering re-releasing the latest as 1.6 and deleting the 2.x releases so as to avoid this for others moving forward...

@francislavoie @samcro1967 would this cause you guys any issues if I were to do that?

**Comment by samcro1967** [comment]
No issue on my end.  Just a simple change in my Dockerfile and app version monitoring software.

**Comment by francislavoie** [comment]
It doesn't really matter to me what option you choose, I'm just letting you know the options.

**Comment by kirsch33** [comment]
Gotcha. Done, and re-released as 1.6.0 

apologies for the confusion. @samcro1967 please follow up once more to let me know this works OK, and i'll close this issue

**Comment by samcro1967** [comment]
Reverting CaddyFile back to the original `--with github.com/kirsch33/realip` I am seeing it grabbing v1.5.4.
```
2022/05/27 11:08:21 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/kirsch33/realip github.com/caddyserver/caddy/v2@v2.5.1 
go: downloading github.com/kirsch33/realip v1.5.4
go: added github.com/kirsch33/realip v1.5.4
```

**Comment by kirsch33** [comment]
@samcro1967 try one more time...sorry, i forgot to remove /v2 from the go.sum file

re-released as 1.6.1 just now. let me know.

**Comment by samcro1967** [comment]
Just tested and is working as expected.
